conans/test/functional/generators/link_order_test.py

```diff
@@ -0,0 +1,341 @@
+import os
+import platform
+import textwrap
+import unittest
+
+from jinja2 import Template
+from parameterized import parameterized
+
+from conans.model.ref import ConanFileReference
+from conans.test.utils.tools import TestClient
+
+
+class LinkOrderTest(unittest.TestCase):
```

Comment:
This test is challenging to understand, possibly hard to maintain.
But I understand it, probably the best is to leave it as-is